src/check.rs

```diff
@@ -143,17 +143,17 @@ fn try_building(
     output.complete_step(version, status.success());
 
     let toolchain = toolchain_specifier.to_owned();
-    let version = version.to_owned();
+    let version = version.clone();
```

Comment:
I see that you replaced most [ToOwned::to_owned](https://doc.rust-lang.org/std/borrow/trait.ToOwned.html) calls with [Clone::clone](https://doc.rust-lang.org/std/clone/trait.Clone.html), and was wondering about your reason for such a change. I think in most cases `to_owned` is the correct call since the reason to clone in these instances is to have ownership,

Reply:
I guess it's a matter of taste. The argument is that the way you get an owned version is by cloning, so using `clone` directly is more explicit. Also it's more consistent in the sense that the method has the same name whether the thing you're trying to get an owned version of is a `&string` or a `&Vec`, or whatever (as opposed to `to_string`, `to_vec`, etc.)

There's a more esoteric advantage too- doing it this way results in less warnings from the default configuration of the clippy 'pedantic' lints.

there's a reference here - https://rust-lang.github.io/rust-clippy/master/index.html#implicit_clone